jazz_codeq/utils.js

```diff
@@ -0,0 +1,276 @@
+		metrics = query.metrics.split(",");
+		
+		if (metrics.length >= 1) {
+			
+			let allowed_metrics = config.ALLOWED_METRICS.map(v => v.toLowerCase());
+			let invalid_metrics = [];
+
+			for (let j = metrics.length - 1; j >= 0; j--) {
+				if (allowed_metrics.indexOf(metrics[j]) === -1) {
+					invalid_metrics.push(metrics[j]);
+				}
+			}
+
+			if (invalid_metrics.length > 0) {
+				let message = messages.MISSING_METRICS + invalid_metrics.join(", ");
+				result = { "error": message, "metrics": []};
+				return result;
+			}
+		} else {
+			result = { "error": messages.INVALID_METRICS + query.metrics, "metrics": []};
+			return result;
+		}
+	} else {
+		result = { "metrics": config.ALLOWED_METRICS };
+	}
+
+	return result;
+}
+
+function replaceKeys(obj, find, replace) {
+	return Object.keys(obj).reduce(
+		(acc, key) => Object.assign(acc, {
+			[key.replace(find, replace)]: obj[key]
+		}), {}
+	);
+}
+
+function getHistoryValues(valuesArray){
+	return valuesArray.map(obj => replaceKeys(obj, 'date', 'ts'));
+}
+
+// get sonar token
+function getSonarToken(config) {
+	return new Promise((resolve, reject) => {
+		const svcPayload = {
+			uri: config.SERVICE_API_URL + config.TOKEN_URL,
+			method: 'POST',
+			json: {
+				"username": config.SONAR_USER,
+				"password": config.SONAR_PASSWORD
+			},
+			rejectUnauthorized: false
+		};
+		logger.info("Getting Sonar token");	
+		request(svcPayload, (error, response, body) => {
+			if (response && response.statusCode === 200 && body && body.data) {
+				resolve(body.data.token);
+			} else {
+				logger.error(response.body.message);
+				reject({
+					"error": "Could not get authentication token.",
+					"message": response.body.message
+				});
+			}
+		});
+	});
+}
+
+//get auth token
+function getEnvironmentToken(config) {
+	return new Promise((resolve, reject) => {
+		const svcPayload = {
+			uri: config.SERVICE_API_URL + config.TOKEN_URL,
+			method: 'POST',
+			json: {
+				"username": config.SERVICE_USER,
+				"password": config.TOKEN_CREDS
+			},
+			rejectUnauthorized: false
+		};
+		
+		logger.info("Getting token for calling Environment API...");
+		request(svcPayload, (error, response, body) => {
+			if (response.statusCode === 200 && body && body.data) {
+				resolve({ "auth_token": body.data.token });
+			} else {
+				logger.error(response.body.message);
+				reject({
+					"token_error": "Could not get authentication token for environment lookup.",
+					"message": response.body.message
+				});
+			}
+		});
+	});
+}
+
+//get branch
+function getProjectBranch(authToken, query, config) {
+	return new Promise((resolve, reject) => {
+		const envID = query.environment;
+		const svcPayload = {
+			uri: config.SERVICE_API_URL + config.ENV_SERVICE + envID + "?domain=" + query.domain + "&service=" + query.service,
+			method: 'GET',
+			headers: {
+				'Authorization': authToken,
+				'Content-Type': 'application/json'
+			},
+			rejectUnauthorized: false
+		};
+		logger.info(svcPayload);
```

Comment:
Same comment as below, these are `debug` logs at best. Also consolidate the two into one.